### ISSUE TITLE
Plus version keeps dropping audio device despite being saved in config

### ISSUE THREAD

**Issue by MxtressMyxx** [milkdrop2077/MilkDrop3#65]
Issue in title

in config:

`AudioDeviceSelected=CABLE Input (VB-Audio Virtual Cable)`

but every time i start milkdrop, it reverts to auto

**Comment by milkdrop2077** [comment]
it reverts to auto in case of an error.
Are you sure the virtual cable is well setup? https://www.patreon.com/posts/how-to-use-your-98362918
did you make it work at least once?

**Comment by MxtressMyxx** [comment]
> it reverts to auto in case of an error. Are you sure the virtual cable is well setup? https://www.patreon.com/posts/how-to-use-your-98362918 did you make it work at least once?

I have used this cable in my stack for years, i don't set it as default or use the listen feature as it would cause other issues down the line in my setup; this is the *only* app that is having an issue with this device

**Comment by Steve4356** [comment]
See https://github.com/milkdrop2077/MilkDrop3/issues/48#issuecomment-2068027376

**Comment by Steve4356** [comment]
> it reverts to auto in case of an error. 

I have a crazy idea: Show the error.

**Comment by milkdrop2077** [comment]
@Steve4356 the error is:

 hr = pAudioClient->Initialize(
     AUDCLNT_SHAREMODE_SHARED,
     AUDCLNT_STREAMFLAGS_LOOPBACK,
     0, 0, pwfx, 0
 );
 if (FAILED(hr)) {
     ERR(L"IAudioClient::Initialize failed: hr = 0x%08x", hr);
     MessageBoxW(NULL, L"MilkDrop3 could not start the Audio Capture :( ...

Basically MilkDrop couldn't initialize the AudioDevice. 
I can't reproduce this error on my computer and updating that part of the code is complex. 
The code is located in [loopback-capture.cpp](https://github.com/milkdrop2077/MilkDrop3/blob/main/code/audio/loopback-capture.cpp)

**Comment by milkdrop2077** [comment]
@MxtressMyxx 

Please download the new MilkDropPanel that will come with v3.26:
https://mega.nz/file/tEEBDB5T#i5gOv4RF12_9rzNBuGl0Fb5hywJ8JzPi3Pk9Y-kRAMs

I've added an 'help' tab and listed most of the common issues and how to solve them.
Let me know if that works for you

**Comment by nklomfas** [comment]
> it reverts to auto in case of an error. Are you sure the virtual cable is well setup? https://www.patreon.com/posts/how-to-use-your-98362918 did you make it work at least once?

Getting the same issue. Just purchased 3.25+. No matter the audio source selected Milkdrop reverts to Auto after restarting. No Error is displayed when it opens. I play my audio through a second channel and not the default source.

**Comment by milkdrop2077** [comment]
@nklomfas did you check the 9 help points in the new MilkDropPanel?

**Comment by Steve4356** [comment]
> @nklomfas did you check the 9 help points in the new MilkDropPanel?

did you check the 9 help points in the new MilkDropPanel?
does that feature work for you?

**Comment by milkdrop2077** [comment]
Yes I can select my 2 pairs of headphones, my USB DAC, TV and I can use the virtual audio cable for my microphone too

**Comment by nklomfas** [comment]
> @nklomfas did you check the 9 help points in the new MilkDropPanel?

Yes, I've run through the 9 points.

I set up the virtual audio cable and ensured that the device was set to 16-bit 48kHz and exclusive mode disabled. I don't have any recording software open and I've disabled all Nahimic services.

It does not matter which input I select. Everytime I reboot the software it goes back to Auto mode.

**Comment by milkdrop2077** [comment]
It's hard to tell, could be some kind of conflict with the Nahimic services not well uninstalled https://www.reddit.com/r/ASUS/comments/asot4q/nahimic_service_causing_problems_all_around/

**Comment by Steve4356** [comment]
I don't think so. I also have ASUS but none of those services installed. Guess it's a case of "works for me" and nothing we can do about it. I've worked around the problem, I just start milkdrop before all those other audio things which for example put the standard audio out device to "valve index headset", and milkdrop is happily staying to listen to the device which was active beforehand.

**Comment by Phr3d13** [comment]
You can add me to the list of people that VB-Cable doesn't work with milkdrop 3. Had it working once, can't get it to work again. With no log available to me, I can't give you any more information. Can verify that VB-Cable works. Milkdrop3 wants nothing to do with it any more though. Went through all the 9 helpful things, nothing works for me any more. Would love to use this as my daily driver for my streams. Kinda sad that I only got to use it once.

**Comment by Phr3d13** [comment]
Managed to get it to work again by selecting VB-Cable as the default windows device, opening milkdrop, then setting the default back to my normal default. very kludgy way to have to make things work. I was trying to switch to get rid of an unnecessary software (winamp) just to have to install another single purpose software to get milkdrop working.

**Comment by Phr3d13** [comment]
I can see paywalling features behind Patreon if things fully worked, but it's like you're just here for the money. Definitely doesn't feel like the way Patreon was intended to be used. If you'd like some help learning github I'm sure myself or someone could get you to a point of confidence. We can't help you if you don't provide us a way to get you useful information.

**Comment by milkdrop2077** [comment]
@Phr3d13 Trust me, if I wanted to make money, I'd be doing something else. Every dollar that I make with md3 is used to take time off from my normal job and develop more features. My audio menu works for 98% of users, and if it doesn't, I can't spend my days answering questions on the internet. 
That part of the code is so technical, it's very difficult to understand what really goes wrong, especially if I don't see it with my own eyes. Feel free to improve the code, though.

**Comment by Phr3d13** [comment]
Can we get current versions of things? What we have is a year old.

**Comment by Phr3d13** [comment]
Or maybe a way to log our errors so you can improve the code? I managed to help a dev at Mixxx add support for my pioneer mixer via the internet with logs and such... Whatever you need to figure out how to capture audio inputs directly instead of needing VB-Cable, I would be happy to supply.
